broken test: `xtrabackup` TestRecovery

This test seems to have regressed and reliably fails on `main`.

Branch: `main`
MySQL version: `mysql-5.7.34-linux-glibc2.12-x86_64`
Xtrabackup version: `percona-xtrabackup-2.4.22-Linux-x86_64.glibc2.12`

Logs:

```
> go test -v .
E0611 17:10:10.666117  354584 topo_process.go:92] Starting etcd with command: etcd --name global --data-dir /home/vmg/src/vitess/vtdataroot/vtroot_41001/topo_41002 --listen-client-urls http://localhost:41002 --advertise-client-urls http://localhost:41002 --initial-advertise-peer-urls http://localhost:41004 --listen-peer-urls http://localhost:41004 --initial-cluster global=http://localhost:41004 --enable-v2=true
=== RUN   TestRecovery
E0611 17:10:32.971006  354584 vtctlclient_process.go:158] Output:
E0611 17:10:32.967821  357716 main.go:67] E0611 15:10:32.967632 backup.go:163] E0611 15:10:32.967551 backup.go:123] backup is not usable, aborting it: [exit status 1
xtrabackup failed with error. Output=xtrabackup: recognized server arguments: --datadir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/data --innodb_data_home_dir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/data --innodb_log_group_home_dir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/logs --log_bin=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/bin-logs/vt-0000009287-bin --server-id=1833938403 --tmpdir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/tmp 
xtrabackup: recognized client arguments: --backup=1 --socket=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/mysql.sock --slave-info=1 --user=vt_dba --target-dir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/tmp --stream=xbstream --password=* 
210611 17:10:31  version_check Connecting to MySQL server with DSN 'dbi:mysql:;mysql_read_default_group=xtrabackup;mysql_socket=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/mysql.sock' as 'vt_dba'  (using password: YES).
Failed to connect to MySQL server as DBD::mysql module is not installed at - line 1327.
210611 17:10:31 Connecting to MySQL server host: localhost, user: vt_dba, password: set, port: not set, socket: /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/mysql.sock
Using server version 5.7.34-log
xtrabackup version 2.4.22 based on MySQL server 5.7.32 Linux (x86_64) (revision id: c99a781)
xtrabackup: uses posix_fadvise().
xtrabackup: cd to /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/data
xtrabackup: open files limit requested 0, set to 1024
xtrabackup: using the following InnoDB configuration:
xtrabackup:   innodb_data_home_dir = /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/data
xtrabackup:   innodb_data_file_path = ibdata1:12M:autoextend
xtrabackup:   innodb_log_group_home_dir = /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/logs
xtrabackup:   innodb_log_files_in_group = 2
xtrabackup:   innodb_log_file_size = 50331648
InnoDB: Number of pools: 1
InnoDB: Last flushed lsn: 2830506 load_index lsn 2760548
InnoDB: An optimized (without redo logging) DDL operation has been performed. All modified pages may not have been flushed to the disk yet.
InnoDB: Last flushed lsn: 2830506 load_index lsn 2792219
InnoDB: An optimized (without redo logging) DDL operation has been performed. All modified pages may not have been flushed to the disk yet.
210611 17:10:31 >> log scanned up to (2830506)
xtrabackup: Generating a list of tablespaces
InnoDB: Allocated tablespace ID 21 for sys/sys_config, old maximum was 0
210611 17:10:32 [01] Streaming /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/data/ibdata1
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./sys/sys_config.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./_vt/shard_metadata.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./_vt/reparent_journal.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./_vt/local_metadata.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./vt_ks/vt_insert_test.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/innodb_index_stats.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/engine_cost.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/slave_relay_log_info.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_category.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/server_cost.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_transition.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_transition_type.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_keyword.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/slave_master_info.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/innodb_table_stats.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_name.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/servers.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_leap_second.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_relation.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/slave_worker_info.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/gtid_executed.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/plugin.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_topic.ibd
210611 17:10:32 [01]        ...done
InnoDB: Last flushed lsn: 2830506 load_index lsn 2859558
InnoDB: An optimized (without redo logging) DDL operation has been performed. All modified pages may not have been flushed to the disk yet. 
PXB will not be able to make a consistent backup. Retry the backup operation
]
Backup Error: rpc error: code = Unknown desc = TabletManager.Backup on zone1-0000009287 error: xtrabackup failed with error. Output=xtrabackup: recognized server arguments: --datadir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/data --innodb_data_home_dir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/data --innodb_log_group_home_dir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/logs --log_bin=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/bin-logs/vt-0000009287-bin --server-id=1833938403 --tmpdir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/tmp 
xtrabackup: recognized client arguments: --backup=1 --socket=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/mysql.sock --slave-info=1 --user=vt_dba --target-dir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/tmp --stream=xbstream --password=* 
210611 17:10:31  version_check Connecting to MySQL server with DSN 'dbi:mysql:;mysql_read_default_group=xtrabackup;mysql_socket=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/mysql.sock' as 'vt_dba'  (using password: YES).
Failed to connect to MySQL server as DBD::mysql module is not installed at - line 1327.
210611 17:10:31 Connecting to MySQL server host: localhost, user: vt_dba, password: set, port: not set, socket: /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/mysql.sock
Using server version 5.7.34-log
xtrabackup version 2.4.22 based on MySQL server 5.7.32 Linux (x86_64) (revision id: c99a781)
xtrabackup: uses posix_fadvise().
xtrabackup: cd to /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/data
xtrabackup: open files limit requested 0, set to 1024
xtrabackup: using the following InnoDB configuration:
xtrabackup:   innodb_data_home_dir = /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/data
xtrabackup:   innodb_data_file_path = ibdata1:12M:autoextend
xtrabackup:   innodb_log_group_home_dir = /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/logs
xtrabackup:   innodb_log_files_in_group = 2
xtrabackup:   innodb_log_file_size = 50331648
InnoDB: Number of pools: 1
InnoDB: Last flushed lsn: 2830506 load_index lsn 2760548
InnoDB: An optimized (without redo logging) DDL operation has been performed. All modified pages may not have been flushed to the disk yet.
InnoDB: Last flushed lsn: 2830506 load_index lsn 2792219
InnoDB: An optimized (without redo logging) DDL operation has been performed. All modified pages may not have been flushed to the disk yet.
210611 17:10:31 >> log scanned up to (2830506)
xtrabackup: Generating a list of tablespaces
InnoDB: Allocated tablespace ID 21 for sys/sys_config, old maximum was 0
210611 17:10:32 [01] Streaming /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/data/ibdata1
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./sys/sys_config.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./_vt/shard_metadata.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./_vt/reparent_journal.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./_vt/local_metadata.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./vt_ks/vt_insert_test.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/innodb_index_stats.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/engine_cost.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/slave_relay_log_info.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_category.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/server_cost.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_transition.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_transition_type.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_keyword.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/slave_master_info.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/innodb_table_stats.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_name.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/servers.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_leap_second.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_relation.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/slave_worker_info.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/gtid_executed.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/plugin.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_topic.ibd
210611 17:10:32 [01]        ...done
InnoDB: Last flushed lsn: 2830506 load_index lsn 2859558
InnoDB: An optimized (without redo logging) DDL operation has been performed. All modified pages may not have been flushed to the disk yet. 
PXB will not be able to make a consistent backup. Retry the backup operation
: exit status 1: xtrabackup failed with error. Output=xtrabackup: recognized server arguments: --datadir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/data --innodb_data_home_dir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/data --innodb_log_group_home_dir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/logs --log_bin=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/bin-logs/vt-0000009287-bin --server-id=1833938403 --tmpdir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/tmp 
xtrabackup: recognized client arguments: --backup=1 --socket=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/mysql.sock --slave-info=1 --user=vt_dba --target-dir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/tmp --stream=xbstream --password=* 
210611 17:10:31  version_check Connecting to MySQL server with DSN 'dbi:mysql:;mysql_read_default_group=xtrabackup;mysql_socket=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/mysql.sock' as 'vt_dba'  (using password: YES).
Failed to connect to MySQL server as DBD::mysql module is not installed at - line 1327.
210611 17:10:31 Connecting to MySQL server host: localhost, user: vt_dba, password: set, port: not set, socket: /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/mysql.sock
Using server version 5.7.34-log
xtrabackup version 2.4.22 based on MySQL server 5.7.32 Linux (x86_64) (revision id: c99a781)
xtrabackup: uses posix_fadvise().
xtrabackup: cd to /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/data
xtrabackup: open files limit requested 0, set to 1024
xtrabackup: using the following InnoDB configuration:
xtrabackup:   innodb_data_home_dir = /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/data
xtrabackup:   innodb_data_file_path = ibdata1:12M:autoextend
xtrabackup:   innodb_log_group_home_dir = /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/logs
xtrabackup:   innodb_log_files_in_group = 2
xtrabackup:   innodb_log_file_size = 50331648
InnoDB: Number of pools: 1
InnoDB: Last flushed lsn: 2830506 load_index lsn 2760548
InnoDB: An optimized (without redo logging) DDL operation has been performed. All modified pages may not have been flushed to the disk yet.
InnoDB: Last flushed lsn: 2830506 load_index lsn 2792219
InnoDB: An optimized (without redo logging) DDL operation has been performed. All modified pages may not have been flushed to the disk yet.
210611 17:10:31 >> log scanned up to (2830506)
xtrabackup: Generating a list of tablespaces
InnoDB: Allocated tablespace ID 21 for sys/sys_config, old maximum was 0
210611 17:10:32 [01] Streaming /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/data/ibdata1
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./sys/sys_config.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./_vt/shard_metadata.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./_vt/reparent_journal.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./_vt/local_metadata.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./vt_ks/vt_insert_test.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/innodb_index_stats.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/engine_cost.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/slave_relay_log_info.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_category.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/server_cost.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_transition.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_transition_type.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_keyword.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/slave_master_info.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/innodb_table_stats.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_name.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/servers.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_leap_second.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_relation.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/slave_worker_info.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/gtid_executed.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/plugin.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_topic.ibd
210611 17:10:32 [01]        ...done
InnoDB: Last flushed lsn: 2830506 load_index lsn 2859558
InnoDB: An optimized (without redo logging) DDL operation has been performed. All modified pages may not have been flushed to the disk yet. 
PXB will not be able to make a consistent backup. Retry the backup operation
: exit status 1
E0611 17:10:32.970059  357716 main.go:76] remote error: rpc error: code = Unknown desc = TabletManager.Backup on zone1-0000009287 error: xtrabackup failed with error. Output=xtrabackup: recognized server arguments: --datadir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/data --innodb_data_home_dir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/data --innodb_log_group_home_dir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/logs --log_bin=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/bin-logs/vt-0000009287-bin --server-id=1833938403 --tmpdir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/tmp 
xtrabackup: recognized client arguments: --backup=1 --socket=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/mysql.sock --slave-info=1 --user=vt_dba --target-dir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/tmp --stream=xbstream --password=* 
210611 17:10:31  version_check Connecting to MySQL server with DSN 'dbi:mysql:;mysql_read_default_group=xtrabackup;mysql_socket=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/mysql.sock' as 'vt_dba'  (using password: YES).
Failed to connect to MySQL server as DBD::mysql module is not installed at - line 1327.
210611 17:10:31 Connecting to MySQL server host: localhost, user: vt_dba, password: set, port: not set, socket: /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/mysql.sock
Using server version 5.7.34-log
xtrabackup version 2.4.22 based on MySQL server 5.7.32 Linux (x86_64) (revision id: c99a781)
xtrabackup: uses posix_fadvise().
xtrabackup: cd to /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/data
xtrabackup: open files limit requested 0, set to 1024
xtrabackup: using the following InnoDB configuration:
xtrabackup:   innodb_data_home_dir = /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/data
xtrabackup:   innodb_data_file_path = ibdata1:12M:autoextend
xtrabackup:   innodb_log_group_home_dir = /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/logs
xtrabackup:   innodb_log_files_in_group = 2
xtrabackup:   innodb_log_file_size = 50331648
InnoDB: Number of pools: 1
InnoDB: Last flushed lsn: 2830506 load_index lsn 2760548
InnoDB: An optimized (without redo logging) DDL operation has been performed. All modified pages may not have been flushed to the disk yet.
InnoDB: Last flushed lsn: 2830506 load_index lsn 2792219
InnoDB: An optimized (without redo logging) DDL operation has been performed. All modified pages may not have been flushed to the disk yet.
210611 17:10:31 >> log scanned up to (2830506)
xtrabackup: Generating a list of tablespaces
InnoDB: Allocated tablespace ID 21 for sys/sys_config, old maximum was 0
210611 17:10:32 [01] Streaming /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/data/ibdata1
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./sys/sys_config.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./_vt/shard_metadata.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./_vt/reparent_journal.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./_vt/local_metadata.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./vt_ks/vt_insert_test.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/innodb_index_stats.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/engine_cost.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/slave_relay_log_info.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_category.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/server_cost.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_transition.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_transition_type.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_keyword.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/slave_master_info.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/innodb_table_stats.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_name.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/servers.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_leap_second.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_relation.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/slave_worker_info.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/gtid_executed.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/plugin.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_topic.ibd
210611 17:10:32 [01]        ...done
InnoDB: Last flushed lsn: 2830506 load_index lsn 2859558
InnoDB: An optimized (without redo logging) DDL operation has been performed. All modified pages may not have been flushed to the disk yet. 
PXB will not be able to make a consistent backup. Retry the backup operation
: exit status 1: xtrabackup failed with error. Output=xtrabackup: recognized server arguments: --datadir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/data --innodb_data_home_dir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/data --innodb_log_group_home_dir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/logs --log_bin=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/bin-logs/vt-0000009287-bin --server-id=1833938403 --tmpdir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/tmp 
xtrabackup: recognized client arguments: --backup=1 --socket=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/mysql.sock --slave-info=1 --user=vt_dba --target-dir=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/tmp --stream=xbstream --password=* 
210611 17:10:31  version_check Connecting to MySQL server with DSN 'dbi:mysql:;mysql_read_default_group=xtrabackup;mysql_socket=/home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/mysql.sock' as 'vt_dba'  (using password: YES).
Failed to connect to MySQL server as DBD::mysql module is not installed at - line 1327.
210611 17:10:31 Connecting to MySQL server host: localhost, user: vt_dba, password: set, port: not set, socket: /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/mysql.sock
Using server version 5.7.34-log
xtrabackup version 2.4.22 based on MySQL server 5.7.32 Linux (x86_64) (revision id: c99a781)
xtrabackup: uses posix_fadvise().
xtrabackup: cd to /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/data
xtrabackup: open files limit requested 0, set to 1024
xtrabackup: using the following InnoDB configuration:
xtrabackup:   innodb_data_home_dir = /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/data
xtrabackup:   innodb_data_file_path = ibdata1:12M:autoextend
xtrabackup:   innodb_log_group_home_dir = /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/logs
xtrabackup:   innodb_log_files_in_group = 2
xtrabackup:   innodb_log_file_size = 50331648
InnoDB: Number of pools: 1
InnoDB: Last flushed lsn: 2830506 load_index lsn 2760548
InnoDB: An optimized (without redo logging) DDL operation has been performed. All modified pages may not have been flushed to the disk yet.
InnoDB: Last flushed lsn: 2830506 load_index lsn 2792219
InnoDB: An optimized (without redo logging) DDL operation has been performed. All modified pages may not have been flushed to the disk yet.
210611 17:10:31 >> log scanned up to (2830506)
xtrabackup: Generating a list of tablespaces
InnoDB: Allocated tablespace ID 21 for sys/sys_config, old maximum was 0
210611 17:10:32 [01] Streaming /home/vmg/src/vitess/vtdataroot/vtroot_41001/vt_0000009287/innodb/data/ibdata1
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./sys/sys_config.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./_vt/shard_metadata.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./_vt/reparent_journal.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./_vt/local_metadata.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./vt_ks/vt_insert_test.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/innodb_index_stats.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/engine_cost.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/slave_relay_log_info.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_category.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/server_cost.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_transition.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_transition_type.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_keyword.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/slave_master_info.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/innodb_table_stats.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_name.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/servers.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone_leap_second.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_relation.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/slave_worker_info.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/gtid_executed.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/plugin.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/time_zone.ibd
210611 17:10:32 [01]        ...done
210611 17:10:32 [01] Streaming ./mysql/help_topic.ibd
210611 17:10:32 [01]        ...done
InnoDB: Last flushed lsn: 2830506 load_index lsn 2859558
InnoDB: An optimized (without redo logging) DDL operation has been performed. All modified pages may not have been flushed to the disk yet. 
PXB will not be able to make a consistent backup. Retry the backup operation
: exit status 1
    recovery.go:190: 
        	Error Trace:	recovery.go:190
        	Error:      	Received unexpected error:
        	           	exit status 1
        	Test:       	TestRecovery
    recovery.go:193: 
        	Error Trace:	recovery.go:193
        	Error:      	Not equal: 
        	           	expected: 0
        	           	actual  : 1
        	Test:       	TestRecovery
--- FAIL: TestRecovery (13.64s)
FAIL
FAIL	vitess.io/vitess/go/test/endtoend/recovery/xtrabackup	34.570s
FAIL
```

Note that by default the full output of `xtrabackup` is not passed through verbatim. A small change is required to obtain the full `stderr` of the process that shows the actual error, which AFAICT is:

> InnoDB: An optimized (without redo logging) DDL operation has been performed. All modified pages may not have been flushed to the disk yet. 
> PXB will not be able to make a consistent backup. Retry the backup operation

--- a/go/test/endtoend/recovery/unshardedrecovery/recovery.go
+++ b/go/test/endtoend/recovery/unshardedrecovery/recovery.go
@@ -100,6 +100,11 @@ func TestMainImpl(m *testing.M) {
 		sql := string(initDb)
 		newInitDBFile = path.Join(localCluster.TmpDirectory, "init_db_with_passwords.sql")
 		sql = sql + initialsharding.GetPasswordUpdateSQL(localCluster)
+		// https://github.com/vitessio/vitess/issues/8315
+		oldAlterTableMode := `
+SET GLOBAL old_alter_table = ON;
+`
+		sql = sql + oldAlterTableMode
 		ioutil.WriteFile(newInitDBFile, []byte(sql), 0666)
 
 		extraArgs := []string{"-db-credentials-file", dbCredentialFile}

--- a/go/vt/mysqlctl/xtrabackupengine.go
+++ b/go/vt/mysqlctl/xtrabackupengine.go
@@ -299,6 +299,7 @@ func (be *XtrabackupEngine) backupFiles(ctx context.Context, params BackupParams
 	// the replication position. Note that if we don't read stderr as we go, the
 	// xtrabackup process gets blocked when the write buffer fills up.
 	stderrBuilder := &strings.Builder{}
+	posBuilder := &strings.Builder{}
 	stderrDone := make(chan struct{})
 	go func() {
 		defer close(stderrDone)
@@ -318,7 +319,7 @@ func (be *XtrabackupEngine) backupFiles(ctx context.Context, params BackupParams
 				}
 				capture = true
 			}
-			fmt.Fprintln(stderrBuilder, line)
+			fmt.Fprintln(posBuilder, line)
 		}
 		if err := scanner.Err(); err != nil {
 			params.Logger.Errorf("error reading from xtrabackup stderr: %v", err)
@@ -359,10 +360,11 @@ func (be *XtrabackupEngine) backupFiles(ctx context.Context, params BackupParams
 	sterrOutput := stderrBuilder.String()
 
 	if err := backupCmd.Wait(); err != nil {
-		return replicationPosition, vterrors.Wrap(err, "xtrabackup failed with error")
+		return replicationPosition, vterrors.Wrap(err, fmt.Sprintf("xtrabackup failed with error. Output=%s", sterrOutput))
 	}
 
-	replicationPosition, rerr := findReplicationPosition(sterrOutput, flavor, params.Logger)
+	posOutput := posBuilder.String()
+	replicationPosition, rerr := findReplicationPosition(posOutput, flavor, params.Logger)
 	if rerr != nil {
 		return replicationPosition, vterrors.Wrap(rerr, "backup failed trying to find replication position")
 	}

